src/group_theory/subgroup.lean

```diff
@@ -404,13 +406,15 @@ end
 @[to_additive] instance fintype_bot : fintype (⊥ : subgroup G) := ⟨{1},
 by {rintro ⟨x, ⟨hx⟩⟩, exact finset.mem_singleton_self _}⟩
 
-@[simp] lemma _root_.add_subgroup.card_bot : fintype.card (⊥ : add_subgroup A) = 1 :=
-fintype.card_eq_one_iff.2
+@[simp] lemma _root_.add_subgroup.card_bot {h : fintype.{v} (⊥ : add_subgroup A)} :
+  @fintype.card (⊥ : add_subgroup A) h = 1 :=
+(@fintype.card_eq_one_iff (⊥ : add_subgroup A) h).2
   ⟨⟨(0 : A), set.mem_singleton 0⟩, λ ⟨y, hy⟩, subtype.eq $ add_subgroup.mem_bot.1 hy⟩
```

Comment:
Am I missing something here? Is this lemma different to what it was before?

Reply:
I added a fintype instance as an argument, since you often end up with different instances in your goal e.g. if you start with `card H = 1` and have a hypothesis `H = bot`

Reply:
You can eliminate all the `@`s and universe parameters with
```
@[simp] lemma _root_.add_subgroup.card_bot [fintype ↥(⊥ : add_subgroup A)] :
  fintype.card (⊥ : add_subgroup A) = 1 :=
(fintype.card_eq_one_iff).2
  ⟨⟨(0 : A), set.mem_singleton 0⟩, λ ⟨y, hy⟩, subtype.eq $ add_subgroup.mem_bot.1 hy⟩
```